src/libcore/alloc.rs

```diff
@@ -241,12 +241,10 @@ impl Layout {
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
     #[inline]
     pub fn repeat(&self, n: usize) -> Result<(Self, usize), LayoutErr> {
-        // This cannot overflow. Quoting from the invariant of Layout:
-        // > `size`, when rounded up to the nearest multiple of `align`,
-        // > must not overflow (i.e., the rounded value must be less than
-        // > `usize::MAX`)
-        let padded_size = self.size() + self.padding_needed_for(self.align());
-        let alloc_size = padded_size.checked_mul(n).ok_or(LayoutErr { private: () })?;
+        let padded_size = self.size().checked_add(self.padding_needed_for(self.align()))
+            .ok_or(LayoutErr { private: () })?;
+        let alloc_size = padded_size.checked_mul(n)
+            .ok_or(LayoutErr { private: () })?;
```

Comment:
Could you add a comment referencing #69225 noting what the problem was?

Reply:
Yes I will do that. I wanted to also explore whether the other comment referenced in the original PR is incorrect, and then ideally add a test. But perhaps for a backport fix just fixing and commenting is best

Reply:
Adding a test in addition to the comment would be ideal if that is possible.

Reply:
Yeah I am quite curious what is wrong about this comment that you are removing -- there are, I think, a few other cases of similar comments in that file.

Reply:
I removed it just because it was added with the commit introducing this issue